### DISCUSSION TITLE
Hide Warning about Automatic Static Optimization

### DISCUSSION THREAD

**Discussion by wuifdesign** [vercel/next.js#35950] [Ideas]
### Describe the feature you'd like to request

A setting to hide following warning:

```
Warning: You have opted-out of Automatic Static Optimization due to `getInitialProps` in `pages/_app`. This does not opt-out pages with `getStaticProps`
Read more: https://nextjs.org/docs/messages/opt-out-auto-static-optimization
```

https://github.com/vercel/next.js/blob/4a7ab34baf932c102fbf281660f09f7eb98c4b74/packages/next/build/index.ts#L1141

I know i use it and i use it because i have to as i need to get settings from api on each call. So i just want to disable this warning. For me, too many warnings during build just hides the relevant warnings.

### Describe the solution you'd like

Maybe a setting in `next.config.js` for all warnings to disable them. Maybe it can also be used to throw an error instead of warning if wanted (if someone wants to):

```
const nextConfig = {
  warnings: {
    staticOptimization: 'off' | 'warn' | 'error',
    ...
  }
}
```

### Describe alternatives you've considered

it may also be changed using a setting in the `.env` file

**Comment by wuifdesign** [comment]
@ijjk there is allready a discussion for this: https://github.com/vercel/next.js/discussions/35938 i made a feature request as i think its a feature request

**Reply by ijjk** [reply]
As mentioned in the feature request template we convert them to discussions now:

> Feature requests will be converted to the GitHub Discussions "Ideas" section.

**Reply by wuifdesign** [reply]
i see. thank you for the information
